src/components/smartSearch/filters/PersonData/index.tsx

```diff
@@ -0,0 +1,195 @@
+/* eslint-disable react-hooks/exhaustive-deps */
+import { FormattedMessage as Msg } from 'react-intl';
+import { Box, Button, Divider, MenuItem, Typography } from '@material-ui/core';
+import { FormEvent, useEffect, useState } from 'react';
+
+import StyledSelect from '../../inputs/StyledSelect';
+import StyledTextInput from '../../inputs/StyledTextInput';
+import useSmartSearchFilter from 'hooks/useSmartSearchFilter';
+import { DATA_FIELD , NewSmartSearchFilter, OPERATION, PersonDataFilterConfig, SmartSearchFilterWithId, ZetkinSmartSearchFilter } from 'types/smartSearch';
+
+const DEFAULT_VALUE = 'none';
+
+interface PersonDataProps {
+    filter:  SmartSearchFilterWithId<PersonDataFilterConfig> |  NewSmartSearchFilter ;
+    onSubmit: (filter: SmartSearchFilterWithId<PersonDataFilterConfig> | ZetkinSmartSearchFilter<PersonDataFilterConfig>) => void;
+    onCancel: () => void;
+}
+
+interface Criterion {
+    field: DATA_FIELD;
+    value: string;
+}
+
+const PersonData = ({ onSubmit, onCancel, filter: initialFilter }: PersonDataProps): JSX.Element => {
+    const { filter, setConfig, setOp } = useSmartSearchFilter<PersonDataFilterConfig>(initialFilter, { fields: { first_name: '' } });
+
+    const ALL_FIELDS = Object.values(DATA_FIELD);
+
+    const initialCriteria = ALL_FIELDS
+        .filter(f => f in filter.config.fields)
+        .map(f => ({ field: f, value: filter.config.fields[f] || '' }));
+
+    const [criteria, setCriteria] = useState<Criterion[]>(initialCriteria);
+
+    const isSubmittable = criteria.every(c => c.value.length);
+
+    useEffect(() => {
+        const fields = criteria.reduce((acc, c) => {
+            return { ...acc, [c.field]: c.value };
+        }, {});
+        setConfig({
+            ...filter.config,
+            fields,
+        });
+    }, [criteria]);
+
+    const addCriteria = (field: string) => {
+        if (field !== DEFAULT_VALUE) {
+            setCriteria([...criteria, { field: field as DATA_FIELD, value: '' }]);
+        }
+    };
+
+    // event handlers
+    const handleAddFilter = (e: FormEvent) => {
+        e.preventDefault();
+        onSubmit(filter);
+    };
+
+    const handleValueChange = (value: string, c: Criterion) => {
+        setCriteria(criteria.map(criterion => {
+            return criterion.field === c.field ?
+                { ...criterion, value } : criterion;
+        }));
+    };
+
+    const handleSelectChange = (field: string, c: Criterion) => {
+        if (field === DEFAULT_VALUE) {
```

Comment:
It was not clear to me when reading this that `DEFAULT_VALUE` was the option that is intended to delete a criterion.

The following two things are true about this constant:

1. It's value is the one that is selected by default in the select box
2. It's value is the one that is represented to users as "remove"

These are two attributes of this constant that are equally true, and could both be reflected in it's name. When naming this constant, the name can draw from either one of these two facts, i.e. it can be something like `DEFAULT_VALUE` or something like `REMOVE_OPTION`. Both of these will be obvious in one context and less obvious in the other:

1. Using a name like `DEFAULT_VALUE` raises no eyebrows when it's passed as the default value to a select box, but is less obvious in this context right here
2. Using a name like `REMOVE_OPTION` makes this use right here very clear, but is less inline with passing it as a default value

In lieu of using a name like `DEFAULT_VALUE_THAT_IS_ALSO_REMOVE_OPTION`, this is a necessary tradeoff to make. The name will either make obvious sense in one context, or in another, but likely not both.

So when picking a name, I think it's wise to go with the one which will add info/make sense in the context that is most difficult to read otherwise. In this case, I would argue that the more difficult-to-read context is the one right here, i.e. where an option is removed, and that the constant should thus be renamed to something like `REMOVE_OPTION` (not married to that specific name of course).

Passing the `REMOVE_OPTION` as the _de facto default value_ to the `value` property of the `StyledSelect` will very clearly communicate to a reader that the _remove_ option is the _default_ one.

More so than checking for `DEFAULT_VALUE` before doing some sort of filtering operation (which at closer glance turns out the be a delete). Why should picking the default value result in deleting?

So, bottom line is this: I think you should either rename this constant, or add some comments here that explains why it's used the way it is here, or both. 😊

Reply:
v. good point! when you put it this way it's hard to un-see